build/configuration-properties-collector-processor/src/main/java/org/hibernate/search/configuration/properties/collector/impl/AsciiDocWriter.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.configuration.properties.collector.impl;
+
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import java.util.function.BiConsumer;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class AsciiDocWriter implements BiConsumer<Map<String, ConfigurationProperty>, Writer> {
+
+	private final Predicate<Map.Entry<String, ConfigurationProperty>> filter;
+
+	public AsciiDocWriter(Predicate<Map.Entry<String, ConfigurationProperty>> filter) {
+		this.filter = filter;
+	}
+
+	@Override
+	public void accept(Map<String, ConfigurationProperty> propertyMap, Writer writer) {
+		List<Map.Entry<String, ConfigurationProperty>> entries = propertyMap.entrySet().stream()
+				.filter( filter )
+				.collect( Collectors.toList() );
+		if ( entries.isEmpty() ) {
+			// nothing to write - return fast.
+			return;
+		}
+
+		Map<String, Collection<ConfigurationProperty>> groups = entries.stream()
+				.map( Map.Entry::getValue )
+				.collect( Collectors.groupingBy( ConfigurationProperty::moduleName, TreeMap::new, Collectors.toCollection(
+						TreeSet::new ) ) );
```

Comment:
I don't think there's a reason for the intermediary list? Or were you working around some compiler limitation?

```suggestion
		List<Map.Entry<String, ConfigurationProperty>> entries = propertyMap.entrySet().stream()
				.filter( filter )
				.map( Map.Entry::getValue )
				.collect( Collectors.groupingBy( ConfigurationProperty::moduleName, TreeMap::new, Collectors.toCollection(
						TreeSet::new ) ) );
		if ( groups.isEmpty() ) {
			// nothing to write - return fast.
			return;
		}
```